test/test_geocaching.py

```diff
@@ -105,6 +108,68 @@ def test_logout(self):
         self.g.logout()
         self.assertIsNone(self.g.get_logged_user())
 
+    def test_load_credentials(self):
+        filename_backup = self.g._credentials_file
+        credentials = {"username": _username, "password": _password}
+        empty_valid_json = {}
+        nonsense_str = b"ss{}ef"        
+
+        with self.subTest("Try to load nonexistent file from current directory"):
+            self.g._credentials_file = "this_file_doesnt_exist.json"
+            with self.assertRaises(LoginFailedException):
+                username, password = self.g._load_credentials()
+
+        with self.subTest("Try to load valid credentials from current directory"):
+            valid = tempfile.NamedTemporaryFile(suffix='.json', prefix='gc_cred_valid_', dir='./', delete=False)
+            valid.write(json.dumps(credentials).encode())
+            valid.close()
+            self.g._credentials_file = os.path.basename(valid.name)
+            username, password = self.g._load_credentials()
+            self.assertEqual(_username, username)
+            self.assertEqual(_password, password)
+        try:
+            os.remove(valid.name)
+        except:
+            pass
```

Comment:
Why these constructions? I think that this command should not fail. If it fails, it's also a test failure.
So could you please move all `os.remove` command into the subtest without try block? Or am I missing something?